src/Symfony/Component/DependencyInjection/Compiler/RegisterServiceSubscribersPass.php

```diff
@@ -85,7 +85,7 @@ protected function processValue($value, $isRoot = false)
                 $serviceMap[$key] = new Reference($type);
             }
 
-            $subscriberMap[$key] = new TypedReference((string) $serviceMap[$key], $type, $declaringClass, $optionalBehavior ?: ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE);
+            $subscriberMap[$key] = new TypedReference($key, $type, $declaringClass, $optionalBehavior ?: ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE);
```

Comment:
This looks wrong to me. Now you are not requesting a type anymore, but a service.

Reply:
yes, that's what I understand from the interface description

Reply:
> ServiceSubscriberInterface is here to ask for a mapping between a local name and a type, not a local name and a name in the main container.

see https://github.com/symfony/symfony/issues/25561#issuecomment-353049420

Reply:
okay I understand when reading the tests. It should definitely be documented